mods/default/tools.lua

```diff
@@ -330,3 +330,70 @@ minetest.register_tool("default:sword_diamond", {
 		damage_groups = {fleshy=8},
 	}
 })
+
+minetest.register_tool("default:skeleton_key", {
+	description = "Skeleton Key",
+	inventory_image = "default_key_skeleton.png",
+	groups = {key = 1},
+	on_place = function(itemstack, placer, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			return nil
+		end
+
+		local pos = pointed_thing.under
+		local node = minetest.get_node(pos)
+
+		if not node or node.name == "ignore" then
+			return nil
+		end
+
+		local on_skeleton_key_use = minetest.registered_nodes[node.name].on_skeleton_key_use
+		if on_skeleton_key_use then
+			-- make a new key secret in case the node callback needs it
+			local newsecret = string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
+			newsecret = newsecret .. string.format("%04x", math.random(65535))
+
+			local secret, name, owner = on_skeleton_key_use(pos, placer, newsecret)
+
+			if secret then
+				-- finish and return the new key
+				itemstack:take_item()
+				itemstack:add_item("default:key")
+				local meta = itemstack:get_meta()
+				meta:set_string("description", "A key to " .. name .. " owned by " .. owner)
+				meta:set_string("secret", secret)
+				return itemstack
+			end
+		end
+		return nil
+	end
+})
+
+minetest.register_tool("default:key", {
+	description = "Key",
+	inventory_image = "default_key.png",
+	groups = {key = 1, not_in_creative_inventory = 1},
+	stack_max = 1,
+	on_place = function(itemstack, placer, pointed_thing)
+		if pointed_thing.type ~= "node" then
+			return nil
+		end
+
+		local pos = pointed_thing.under
+		local node = minetest.get_node(pos)
+
+		if not node or node.name == "ignore" then
+			return nil
+		end
+
+		local on_key_use = minetest.registered_nodes[node.name].on_key_use
```

Comment:
Also fixed the 2 `nil` returns above which are wrong as well (need to return `itemstack` as well there).